tensorflow_addons/optimizers/novograd.py

```diff
@@ -0,0 +1,246 @@
+        """
+        super(Novograd, self).__init__(name, **kwargs)
+        if weight_decay < 0.0:
+            raise ValueError('Weight decay rate cannot be negative')
+        self._set_hyper('learning_rate', kwargs.get('lr', learning_rate))
+        self._set_hyper('decay', self._initial_decay)
+        self._set_hyper('beta_1', beta_1)
+        self._set_hyper('beta_2', beta_2)
+        self._set_hyper('weight_decay', weight_decay)
+        self._set_hyper('grad_averaging', grad_averaging)
+        self.amsgrad = amsgrad
+        self.epsilon = epsilon or tf.keras.backend.epsilon()
+
+    def _create_slots(self, var_list):
+        # Create slots for the first and second moments.
+        # Separate for-loops to respect the ordering of slot variables from v1.
+        for var in var_list:
+            self.add_slot(var=var, slot_name='m', initializer='zeros')
+        for var in var_list:
+            self.add_slot(
+                var=var,
+                slot_name='v',
+                initializer=tf.zeros(shape=[], dtype=var.dtype))
+        if self.amsgrad:
+            for var in var_list:
+                self.add_slot(var, 'vhat')
+
+    def _prepare_local(self, var_device, var_dtype, apply_state):
+        super(Novograd, self)._prepare_local(var_device, var_dtype,
+                                             apply_state)
+        beta_1_t = tf.identity(self._get_hyper('beta_1', var_dtype))
+        beta_2_t = tf.identity(self._get_hyper('beta_2', var_dtype))
+        apply_state[(var_device, var_dtype)].update(
+            dict(
+                epsilon=tf.convert_to_tensor(self.epsilon, var_dtype),
+                beta_1_t=beta_1_t,
+                beta_2_t=beta_2_t,
+                one_minus_beta_2_t=1 - beta_2_t,
+                one_minus_beta_1_t=1 - beta_1_t,
+            ))
+
+    def set_weights(self, weights):
+        params = self.weights
+        # If the weights are generated by Keras V1 optimizer, it includes vhats
+        # even without amsgrad, i.e, V1 optimizer has 3x + 1 variables, while V2
+        # optimizer has 2x + 1 variables. Filter vhats out for compatibility.
+        num_vars = int((len(params) - 1) / 2)
+        if len(weights) == 3 * num_vars + 1:
+            weights = weights[:len(params)]
+        super(Novograd, self).set_weights(weights)
+
+    def _resource_apply_dense(self, grad, var, apply_state=None):
+        var_device, var_dtype = var.device, var.dtype.base_dtype
+        coefficients = ((apply_state or {}).get((var_device, var_dtype))
+                        or self._fallback_apply_state(var_device, var_dtype))
+        weight_decay = self._get_hyper('weight_decay')
+        grad_averaging = self._get_hyper('grad_averaging')
+
+        v = self.get_slot(var, 'v')
+        g_2 = tf.reduce_sum(tf.square(tf.cast(grad, tf.float32)))
+        v_t = tf.cond(
+            tf.equal(self.iterations,
+                     0), lambda: g_2, lambda: v * coefficients['beta_2_t'] +
```

Comment:
Is it possible to put `tf.equal(iterations, 0)` in one line, which is more readable.

Reply:
Done.